src/r_things.c

```diff
@@ -512,6 +512,20 @@ void R_ProjectSprite (mobj_t* thing)
   if (abs(tx)>(tz<<2))
     return;
 
+  { // [Alaux] Lock crosshair on linetarget
+    extern boolean hud_crosshair_target;
+    extern mobj_t *linetarget;
+    extern void HU_UpdateCrosshairLock(int x, int y);
+    
+    if (STRICTMODE(hud_crosshair_target) && thing == linetarget)
```

Comment:
How does this work? Is it by chance that line target is set to the player's target in the hud code before we start projecting things? It would be less mysterious to use something like `thing == crosshair_target` I think. If Fabian agrees anyway :^)

Reply:
Well, I believe all `linetarget` "searches" use the same code, so the `linetarget` used here must be the same as the one used in `HU_UpdateCrosshair()` to apply the crosshair target highlight color, not to say the one used for everything else that asks for `linetarget`.

Reply:
The question is if that's a guarantee for the most recent linetarget. If not, this is brittle code that has a good chance of being broken in the future - unless we separately track a crosshair target.

Reply:
Sounds like a good idea to me, but I'd like Fabian's approval first.

Reply:
I agree we should introduce a `mobj_t *crosshair_target` variable and set this to `crosshair_target = linetarget` right after the `P_AimLineAttack()` sqeuence in `HU_UpdateCrosshair()`.

Reply:
Also, could you please move this block down a bit further? There are still various occasions for the `R_ProjectSprite()` function to `return` early and we wouldn't want to update the crosshair corrdinates in that case anyway.

Reply:
In the latest commit, I moved it right before `tx` is modified to "calculate edges of the shape", after which point the crosshair would be drawn at, well, the edges of the shapes.